test/foundry/deploy/11_HolographFactory.t.sol

```diff
@@ -0,0 +1,353 @@
+   */
+  function testDeployRevertSignatureRIncorrect() public {
+    (DeploymentConfig memory deployConfig, bytes32 hashHtokenEth) = getConfigHtokenETH();
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+      privateKeyDeployer,
+      HelperSignEthMessage.toEthSignedMessageHash(hashHtokenEth)
+    );
+    Verification memory signature = Verification({v: v, r: bytes32(invalidSignature), s: s});
+
+    vm.expectRevert(bytes(ErrorConstants.INVALID_SIGNATURE_ERROR_MSG));
+    vm.prank(deployer);
+    holographFactory.deployHolographableContract(deployConfig, signature, deployer);
+  }
+  /**
+   * @notice Test the deployHolographableContract function Revert if the signature S is invalid
+   * @dev  Refers to the hardhat test with the description 'should fail with invalid signature if signature.s is incorrect'
+   */
+  function testDeployRevertSignatureSIncorrect() public {
+    (DeploymentConfig memory deployConfig, bytes32 hashHtokenEth) = getConfigHtokenETH();
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+      privateKeyDeployer,
+      HelperSignEthMessage.toEthSignedMessageHash(hashHtokenEth)
+    );
+    Verification memory signature = Verification({v: uint8(bytes1(invalidSignature)), r: r, s: s});
+
+    vm.expectRevert(bytes(ErrorConstants.INVALID_SIGNATURE_ERROR_MSG));
+    vm.prank(deployer);
+    holographFactory.deployHolographableContract(deployConfig, signature, deployer);
+  }
+  /**
+   * @notice Test the deployHolographableContract function Revert if the signature V is invalid
+   * @dev  Refers to the hardhat test with the description 'should fail with invalid signature if signature.s is incorrect'
+   */
+  function testDeployRevertSignatureVIncorrect() public {
+    (DeploymentConfig memory deployConfig, bytes32 hashHtokenEth) = getConfigHtokenETH();
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+      privateKeyDeployer,
+      HelperSignEthMessage.toEthSignedMessageHash(hashHtokenEth)
+    );
+    Verification memory signature = Verification({v: uint8(bytes1(invalidSignature)), r: r, s: s});
+
+    vm.expectRevert(bytes(ErrorConstants.INVALID_SIGNATURE_ERROR_MSG));
+    vm.prank(deployer);
+    holographFactory.deployHolographableContract(deployConfig, signature, deployer);
+  }
+  /**
+   * @notice Test the deployHolographableContract function Revert if the signer is invalid
+   * @dev  Refers to the hardhat test with the description 'should fail with invalid signature if signer is incorrect'
+   */
+  function testDeployRevertSignatureSignIncorrect() public {
+    (DeploymentConfig memory deployConfig, bytes32 hashHtokenEth) = getConfigHtokenETH();
+
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+      privateKeyDeployer,
+      HelperSignEthMessage.toEthSignedMessageHash(hashHtokenEth)
+    );
+    Verification memory signature = Verification({v: uint8(bytes1(invalidSignature)), r: r, s: s});
```

Comment:
@mmvazzano this should be using a valid signature.

Reply:
Done!
Same as [here](https://github.com/holographxyz/holograph-protocol/pull/325#discussion_r1626657890)